network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py

```diff
@@ -663,7 +682,16 @@ def total_vote_count(self):
 
     @property
     def creepiness(self):
-        return self.evaluation.average_creepiness
+        try:
+            # Try an annotation made above the ProductPage level
+            return self._average_creepiness
+        except AttributeError:
+            try:
+                # It failed, let's try an annotation made at the ProductPageEvaluation level
+                return self.evaluation._average_creepiness
+            except AttributeError:
+                # We don't have an annotation, let's calculate it ourselves
+                return self.evaluation.average_creepiness
```

Comment:
**question:** Is there ever a case where it makes sense to have this fallback? With the DB optimisation available, perhaps this is another case where raising an Exception would help avoid future developers avoid potentially expensive calls?

Reply:
You're right, I'll throw an exception here as well